search: add versions info to search list item

jennur opened the pull request:
Closes https://github.com/zenodo/zenodo-rdm/issues/43

- Adds number of published versions to the search list item (hidden when all versions show in the list)

## Screenshots
<img width="1303" alt="Screenshot 2022-11-21 at 16 48 47" src="https://user-images.githubusercontent.com/21052053/203100572-f1e52a37-20fb-4e97-9214-4de277e2bfa6.png">

### When viewing all versions
<img width="1308" alt="Screenshot 2022-11-21 at 16 48 37" src="https://user-images.githubusercontent.com/21052053/203100594-e1da2153-b80a-457d-8788-c8b74bde539f.png">

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/search/components.js:
```diff
@@ -89,17 +93,26 @@ export const RDMRecordResultsListItem = ({ result }) => {
+            <p>
+              <small>
+                <b>
+                  {versions.index - 1} more version(s) exist for this record
```
Minor: 
```suggestion
                  {versions.index - 1} more {versions.length > 2 ? versions : version} exist for this record
```

jrcastro2 reviewed (commented):
Tests for eslint are failing.

jennur commented:
## Update

- I discovered an issue with the use of `withState` on `RDMRecordResultsListItem`, as it is used both inside and outside of the search app. This is why I split it up and added an extension of the component `RDMRecordResultsListItemWithState`.
- Text changed as suggested by @jrcastro2
### Screenshot 

![Screenshot 2022-11-24 at 09 42 55](https://user-images.githubusercontent.com/21052053/203747287-342d9acb-646c-43ae-8e42-0676abb6d3dd.png)
